site/content/docs/main/csi-snapshot-data-movement.md

```diff
@@ -0,0 +1,391 @@
+Velero's known object storage providers are included here [supported providers][9], for which, Velero pre-defines the endpoints. If you want to use a different backup storage, make sure it is S3 compatible and you provide the correct bucket name and endpoint in BackupStorageLocation. Velero handles the creation of the backup repo prefix in the backup storage, so make sure it is specified in BackupStorageLocation correctly.  
+
+Velero creates one backup repository per namespace. For example, if backing up 2 namespaces, namespace1 and namespace2, using kopia repository on AWS S3, the full backup repo path for namespace1 would be `https://s3-us-west-2.amazonaws.com/bucket/kopia/ns1` and for namespace2 would be `https://s3-us-west-2.amazonaws.com/bucket/kopia/ns2`.  
+
+There may be additional installation steps depending on the cloud provider plugin you are using. You should refer to the [plugin specific documentation][9] for the must up to date information.  
+
+## Install Velero with CSI support on source cluster
+
+On source cluster, Velero needs to manipulate CSI snapshots through the CSI volume snapshot APIs, so you must enable the `EnableCSI` feature flag and install the Velero [CSI plugin][2] on the Velero server.  
+
+Both of these can be added with the `velero install` command.
+
+```bash
+velero install \
+--features=EnableCSI \
+--plugins=<object storage plugin>,velero/velero-plugin-for-csi:v0.6.0 \
+...
+```
+
+### Configure storage class on target cluster
+
+For Velero built-in data movement, CSI facilities are not required necessarily in the target cluster. On the other hand, the volume provision should work similarly to the source cluster. For example, the same storage class should be working in the target cluster.  
+By default, Velero won't restore storage class resources from the backup since they are cluster scope resources. However, if you specify the `--include-cluster-resources` restore flag, they will be restored. For a cross provider scenario, the storage class from the source cluster is probably not usable in the target cluster.  
+In either of the above cases, the best practice is to create a working storage class in the target cluster with the same name as it in the source cluster. In this way, even though `--include-cluster-resources` is specified, Velero restore will skip restoring the storage class since it finds an existing one.  
+Otherwise, if the storage class name in the target cluster is different, you can change the PVC's storage class name during restore by the [changing PV/PVC storage class][10] method. You can also configure to skip restoring the storage class resources from the backup since they are not usable.  
+
+### Customized Data Movers
+
+If you are using a customized data mover, follow the data mover's instructions for any further prerequisites.  
+For Velero side configurations mentioned above, the installation and configuration of node-agent may not be required.  
+
+
+## To back up
```

Comment:
For both backup and restore, shall we mention how to monitor the progress of the data movement of each volume via `velero backup/restore describe --details` ?